components/auth/RegisterInput.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState, useCallback, FC, KeyboardEvent, useContext } from 'react';
+import styled from 'styled-components';
+import { useRouter } from 'next/router';
+import qs from 'qs';
+
+import Section from '@/components/Section';
+import { media } from '@/utils/theme';
+// import { getRelativePath } from '@/utils';
+import TextInput from '@/components/TextInput';
+import Button from '@/components/Button';
+// import { Router } from '@/i18n';
+import { UserContext } from '@/contexts/user';
+// import { signIn } from '@/utils/api';
+// import { preparedUser } from '@/utils/user';
+// import notify from '@/utils/notify';
+// import { CertsContext } from '@/contexts/certs';
+
+
+import Loading from '../Loading';
+
+const StyledSection = styled(Section)`
+  position: absolute;
+  left: 0;
+  top: 0;
+  height: 100vh;
+  > img {
+    position: absolute;
+    left: 0;
+    top: 0;
+    width: 100%;
+    height: 100%;
+  }
+`;
+
+const RegisterWrapper = styled.div`
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  height: 100%;
+  ${media('desktop')} {
+    width: 100%;
+  }
+`;
+
+const StyledTextInput = styled(TextInput)`
+  && {
+    width: 50%;
+    margin-bottom: 60px;
+  }
+`;
+
+const StyledButton = styled(Button)`
+  padding: ;
+  margin-top: 2em;
+`;
+
+const InfoWrapper = styled.div`
+  max-width: 1000px;
+  width: 100%;
+  padding: 40px 40px 72px 40px;
+  background-color: #ffffff;
+  font-weight: bold;
+  > p {
+    color: #a80100;
+    font-size: 4vmin;
+    letter-spacing: 1.7px;
+    display: block;
+    max-width: 397px;
+    width: 100%;
+    margin-bottom: 100px;
+    ${media('desktop')} {
+      font-size: 24px;
+    }
+  }
+`;
+
+const ErrorMessage = styled.p`
+  font-size: ${p => p.theme.fontSize.smaller};
+  color: ${p => p.theme.colors.primary};
+`;
+
+type TProps = {
+  setPageState: string;
+  onChange: (e: any) => void;
+}
+
+
+const InputName: FC<TProps> = (setPageState) => {
+  const { query } = useRouter();
+  const { updateUser } = useContext(UserContext);
+  const [name, setName] = useState<string>('');
+  const [user, setUser] = useState<string>('');
+  const [error, setError] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const onInputName = useCallback(async () => {
+    const validate = () => {}
+
+     // call API
+    const result = 'fakeUser';
+    if (result) {
+      // handle success
+      setUser(result);
+      setPageState.onChange('Signin');
```

Comment:
Because setPageState could include other `props`, please rename `onChange` to `onChangePageState` in the register page .

And here use `props.onChangePageState`.